templates/sample-app/.config.js

```diff
@@ -0,0 +1,232 @@
+ *  OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL 
+ *  THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, 
+ *  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ *  OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+ *  OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ **/
+
+/*
+ * Forked from https://github.com/npm/init-package-json/blob/master/default-input.js
+ * 
+ * We can't simply require this file from init-package-json, and override its exported members. 
+ * It seems to reference vars that don't exist, leading me to believe that the code here is wrapped in something before it is ran.
+ * 
+ * Therefore, we have to copy the entire file over and override the bits we want to change.
+ */
+
+var fs = require('fs')
+var glob = require('glob')
+var path = require('path')
+var validateLicense = require('validate-npm-package-license')
+var validateName = require('validate-npm-package-name')
+var npa = require('npm-package-arg')
+var semver = require('semver')
+
+function isTestPkg(p) {
+	return !!p.match(/^(expresso|mocha|tap|coffee-script|coco|streamline)$/)
+}
+
+function niceName(n) {
+	return n.replace(/^node-|[.-]js$/g, '').toLowerCase()
+}
+
+function conf(name) {
+	return config.get(name) || config.get(name.split('.').join('-'))
+}
+
+var name = package.name || basename
+var spec = npa(name)
+var scope = conf('scope')
+if (scope) {
+	if (scope.charAt(0) !== '@') scope = '@' + scope
+	if (spec.scope) {
+		name = scope + '/' + spec.name.split('/')[1]
+	} else {
+		name = scope + '/' + name
+	}
+}
+exports.name = yes ? name : prompt('package name', niceName(name), function (data) {
+	var its = validateName(data)
+	if (its.validForNewPackages) return data
+	var errors = (its.errors || []).concat(its.warnings || [])
+	var er = new Error('Sorry, ' + errors.join(' and ') + '.')
+	er.notValid = true
+	return er
+})
+
+var version = package.version || conf('init.version') || '1.0.0'
+exports.version = yes ?
+	version :
+	prompt('version', version, function (version) {
+		if (semver.valid(version)) return version
+		var er = new Error('Invalid version: "' + version + '"')
+		er.notValid = true
+		return er
+	})
+
+if (!package.description) {
+	exports.description = yes ? '' : prompt('description')
+}
+
+if (!package.bin) {
+	exports.bin = function (cb) {
+		fs.readdir(path.resolve(dirname, 'bin'), function (er, d) {
+			// no bins
+			if (er) return cb()
+			// just take the first js file we find there, or nada
+			return cb(null, d.filter(function (f) {
+				return f.match(/\.js$/)
+			})[0])
+		})
+	}
+}
+
+exports.directories = function (cb) {
+	fs.readdir(dirname, function (er, dirs) {
```

Comment:
Argh, I don't think there's a line in this function that doesn't have an abbreviation in it.
I can take `cb`, `dirs` and `dirname` as they're fairly standard, but would prefer real names.
I don't like `er`, `d` or `res`.